### DISCUSSION TITLE
Image/Table numbering and labels

### DISCUSSION THREAD

**Discussion by chrisdburr** [squidfunk/mkdocs-material#4673] [Q&A]
Apologies if this is in the documentation already, it's possible I am not searching for the correct terms, and have also struggled to find anything on MK Docs in general. 

## Desired Outcome
To be able to automatically number images and tables, and then have a label that can be used to cross-reference the element in the rest of the docs—either natively or through a plugin. 

## Example
Let's say I have an image (or figure), which is added as a figure, as follows:

```
<figure markdown>
  ![Description of the image](path/to/image.png)
  <figcaption>A caption for the image</figcaption>
</figure>
```

Is it possible to do the following:

a) automatically number this figure, such that the caption reads 'Figure X: A Caption for the image'
b) cross-reference the figure in the documents, using a label such as `See @fig-label for details.`

**Comment by squidfunk** [comment]
This is an excellent use case for a Markdown extension or MkDocs plugin. Have you checked the [MkDocs plugin Wiki](https://github.com/mkdocs/mkdocs/wiki/MkDocs-Plugins)?

**Reply by chrisdburr** [reply]
No. I was not aware of this excellent looking repo. I'll have a look through it later and report back on this discussion post if I find something suitable. Thanks!

**Comment by chrisdburr** [comment]
Some really great looking plugins, but none that meet the desired outcome, unfortunately. Some are in the right ballpark (e.g. [autorefs](https://github.com/mkdocstrings/)), but nothing I can see that would do both automatic numbering and also allow for cross-referencing. 

If anyone has found a workaround for this, I’d appreciate any suggestions.
